src/pkg/cli/client/byoc/do/byoc.go

```diff
@@ -70,15 +75,99 @@ func NewByocClient(ctx context.Context, grpcClient client.GrpcClient, tenantId t
 	return b, nil
 }
 
+func (b *ByocDo) getCdImageTag(ctx context.Context) (string, error) {
+	if b.cdImageTag != "" {
+		return b.cdImageTag, nil
+	}
+
+	projInfo, err := b.getProjectProto(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	// send project update with the current deploy's cd version,
+	// most current version if new deployment
+	deploymentCdImageTag := byoc.CdLatestImageTag
+	if (projInfo != nil) && (len(projInfo.Services) > 0) && (projInfo.CdVersion != "") {
+		deploymentCdImageTag = projInfo.CdVersion
+	}
+
+	// possible values are [public-beta, 1, 2, ...]
+	return deploymentCdImageTag, nil
+}
+
+func annotateAwsError(err error) error {
+	if err == nil {
+		return nil
+	}
+	if strings.Contains(err.Error(), "get credentials:") {
+		return connect.NewError(connect.CodeUnauthenticated, err)
+	}
+	if aws.IsS3NoSuchKeyError(err) {
+		return connect.NewError(connect.CodeNotFound, err)
+	}
+	if aws.IsParameterNotFoundError(err) {
+		return connect.NewError(connect.CodeNotFound, err)
+	}
+	return err
+}
+
+func (b *ByocDo) getProjectProto(ctx context.Context) (*defangv1.ProjectUpdate, error) {
+	client, err := b.driver.CreateS3Client()
+	if err != nil {
+		return nil, err
+	}
+
+	bucketName, err := b.driver.GetBucketName(ctx, client)
+	if err != nil {
+		return nil, err
+	}
+
+	if bucketName == "" {
+		return nil, errors.New("no bucket found")
+	}
+
+	path := fmt.Sprintf("projects/%s/%s/project.pb", b.ProjectName, b.PulumiStack)
+	getObjectOutput, err := client.GetObject(ctx, &s3.GetObjectInput{
+		Bucket: &bucketName,
+		Key:    &path,
+	})
+
+	if err != nil {
+		if aws.IsS3NoSuchKeyError(err) {
+			term.Debug("s3.GetObject:", err)
+			return nil, nil // no services yet
+		}
+		return nil, annotateAwsError(err)
+	}
+	defer getObjectOutput.Body.Close()
+	base64Reader := base64.NewDecoder(base64.StdEncoding, getObjectOutput.Body)
```

Comment:
Necessary decoding but not in AWS side. The AWS marshalling flow is slightly different which may account for the need to to decode and unmarshal rather than just unmarshing.

Reply:
I wonder if this is because of the `content` vs `contentBase64` bug we fixed. We can simply ignore the faulty payloads.